source/common/router/rds_impl.cc

```diff
@@ -206,6 +217,37 @@ void RdsRouteConfigProviderImpl::onConfigUpdate() {
     prev_config->config_ = new_config;
     return previous;
   });
+
+  const auto aliases = config_update_info_->aliasesInLastVhdsUpdate();
```

Comment:
Just daydreaming here, excuse me if there is something about the code flow I don't know that makes this a stupid question. What if there are two VHDS updates before you arrive here to use the first one's aliases? Would they be lost?

Reply:
> What if there are two VHDS updates before you arrive here to use the first one's aliases? Would they be lost?

This is called synchronously in onConfigUpdate callbacks for RDS and VHDS  (https://github.com/envoyproxy/envoy/blob/master/source/common/router/rds_impl.cc#L123 and https://github.com/envoyproxy/envoy/blob/master/source/common/router/vhds.cc#L56 respectively). My understanding is that each DiscoveryResponse will trigger an onConfigUpdate callback which will always complete, even if additional DiscoveryResponses arrive.

Reply:
Got it, yes, that makes sense and sounds solid.